src/components/manage/Widgets/TextWidget.jsx

```diff
@@ -122,7 +122,9 @@ class TextWidget extends Component {
           maxLength={maxLength || null}
         />
         {icon && iconAction && (
-          <button className={`field-${id}-action-button`} onClick={iconAction}>
+          <button className={`field-${id}-action-button`} onClick={(e) => {
+            iconAction(e, id, onChange);
```

Comment:
I think its too specific. I would directly implement iconAction as () => {onChangeField() or onChangeBlock()} in my component.